compiler/src/dotty/tools/dotc/core/Annotations.scala

```diff
@@ -73,7 +73,7 @@ object Annotations {
 
     /** Does this annotation refer to a parameter of `tl`? */
     def refersToParamOf(tl: TermLambda)(using Context): Boolean =
-      val args = arguments
+      val args = tpd.allArguments(tree)
       if args.isEmpty then false
       else tree.existsSubTree:
         case id: (Ident | This) => id.tpe.stripped match
```

Comment:
When I was playing with annotations and macros, I also noticed that this check was not enough: when we write down in source code a singleton type `x.type`, it will be parsed as `SingletonTypeTree(Ident(x))`, but in a macro we usually create a reference to `x.type` with just a `TermRef`, so no `Ident` tree node will be involved.

Reply:
When generated from a macro, wouldn't the generated type always be encapsulated in a `TypeTree`?

Maybe something like that would be sufficient?

```scala
    def refersToParamOf(tl: TermLambda)(using Context): Boolean =
      def referToParam(tp: Type) =
        tp.stripped match
          case TermParamRef(tl1, _) => tl eq tl1
          case _ => false
      val args = tpd.allArguments(tree)
      if args.isEmpty then false
      else tree.existsSubTree:
        case id: (Ident | This) => referToParam(id.tpe)
        case tpt: TypeTree => tpt.tpe.existsPart(referToParam)
        case _ => false
```